packages/ecc-client-ga4gh-tes/src/components/tes-create-run/definition/tesCreateRun.ts

```diff
@@ -0,0 +1,579 @@
+in the [FAST documentation](https://www.fast.design/docs/fast-element/using-directives/#the-repeat-directive:~:text=Some%20context%20properties,template%20from%20above%3A)
+can have performance issues, hence these rules are opted out.
+*/
+import {
+  FASTElement,
+  attr,
+  observable,
+  customElement,
+} from '@microsoft/fast-element';
+import template from './tesCreateRun.template.js';
+import styles from './tesCreateRun.styles.js';
+import CreateTaskData, {
+  ExecutorData,
+  InputData,
+  OutputData,
+} from './createTask.js';
+import { postTask } from '../../../data/Task/tesGet.js';
+
+const executorTemplate: ExecutorData = {
+  command: [],
+  env: {},
+  image: '',
+  stderr: '',
+  stdin: '',
+  stdout: '',
+  workdir: '',
+};
+
+const inputTemplate: InputData = {
+  path: '',
+  url: '',
+};
+
+const outputTemplate: OutputData = {
+  path: '',
+  url: '',
+  type: '',
+};
+@customElement({
+  name: 'ecc-client-ga4gh-tes-create-run',
+  template,
+  styles,
+  shadowOptions: { mode: 'open' },
+})
+export default class TESCreateRun extends FASTElement {
+  @attr baseURL = '';
+
+  @attr name = '';
+
+  @attr state = 'UNKNOWN';
+
+  @attr description = '';
+
+  @attr taskExecutors: ExecutorData[] = [{ ...executorTemplate }];
+
+  @observable taskExecutorsLength = this.taskExecutors.length;
+
+  @attr taskInput: InputData[] = [{ ...inputTemplate }];
+
+  @observable taskInputLength = this.taskInput.length;
+
+  @attr taskOutput: OutputData[] = [{ ...outputTemplate }];
+
+  @observable taskOutputLength = 1;
+
+  @attr cpu_cores = '4';
+
+  @attr disk_gb = '40';
+
+  @attr preemptible = false;
+
+  @attr ram_gb = '8';
+
+  @attr zones: string[] = [];
+
+  @attr WORKFLOW_ID = '';
+
+  @attr PROJECT_GROUP = '';
+
+  @attr volumes: string[] = [];
+
+  @observable response = {};
+
+  @observable taskData: CreateTaskData = {
+    name: this.name,
+    state: this.state,
+    description: this.description,
+    executors: this.taskExecutors,
+    inputs: this.taskInput,
+    outputs: this.taskOutput,
+    resources: {
+      cpu_cores: parseInt(this.cpu_cores, 10),
+      disk_gb: parseInt(this.disk_gb, 10),
+      preemptible: this.preemptible,
+      ram_gb: parseInt(this.ram_gb, 10),
+      zones: this.zones,
+    },
+    tags: {
+      WORKFLOW_ID: this.WORKFLOW_ID,
+      PROJECT_GROUP: this.PROJECT_GROUP,
+    },
+    volumes: this.volumes,
+  };
+
+  connectedCallback(): void {
+    super.connectedCallback();
+    this.taskExecutorsLength = this.taskExecutors.length;
+    this.taskInputLength = this.taskInput.length;
+    this.taskOutputLength = this.taskOutput.length;
+  }
+
+  /**
+   * Handles submit button click
+   */
+  handleSubmit = async () => {
+    // Compute all the task information and create the task schema
+    // <----------------------------------------------------------------->
+    // All the fields input by user are compiled according to task schema
+
+    // Check if all the fields are filled
+    if (
```

Comment:
use a helper function for this. And then you can just iterate through the object and check to make sure all the fields are filled

Reply:
I think I'll need to create a separate issue for this, in the latest discussion, it turns out not all fields are required.

Reply:
okay sounds fine for now